lib/_http_server.js

```diff
@@ -904,8 +904,20 @@ function parserOnIncoming(server, socket, state, req, keepAlive) {
          resOnFinish.bind(undefined,
                           req, res, socket, state, server));
 
-  if (req.headers.expect !== undefined &&
-      (req.httpVersionMajor === 1 && req.httpVersionMinor === 1)) {
+  const isHttp11 = (req.httpVersionMajor === 1 && req.httpVersionMinor === 1);
+
+  // From RFC 2730 5.4 https://datatracker.ietf.org/doc/html/rfc7230#section-5.4
+  // A server MUST respond with a 400 (Bad Request) status code to any
+  // HTTP/1.1 request message that lacks a Host header field and to any
```

Comment:
Maybe we should truncate after "lacks a Host header field" since the rest is irrelevant and isn't checked for here?

Reply:
fixed~